ts/javascript/initializers.ts

```diff
@@ -0,0 +1,207 @@
+import { Initializer } from "../base.ts";
+import * as fs from "https://deno.land/std@0.201.0/fs/mod.ts";
+import { assertEquals } from "https://deno.land/std@0.201.0/assert/mod.ts";
+import outdent from "https://deno.land/x/outdent@v0.8.0/mod.ts";
+import { mapValues } from "../internal/utils.ts";
+
+const npmInitializer: Initializer = () => {
+  const existsPkgJson = fs.existsSync("package.json");
+  const existsPkgLock = fs.existsSync("package-lock.json");
+  const existsYarnLock = fs.existsSync("yarn.lock");
+  if (!existsPkgJson || (!existsPkgLock && existsYarnLock)) {
```

Comment:
This omits the case that both package-lock.json and yarn exist, in which case I figured it was better to use the npm initializer, but let me know if you disagree.

For reference this is the inversion of the following logic which is probably easier to read (and maybe that means we should save it to a variable and negate that in the if statement rather than in-lining it):
```typescript
existsPkgJson && (existsPkgLock || !existsYarnLock)
```